src/openrct2-ui/windows/RideConstruction.cpp

```diff
@@ -1933,14 +1933,15 @@ static void window_ride_construction_mouseup_demolish(rct_window* w)
         _currentTrackPieceType, 0,
         { _currentTrackBegin.x, _currentTrackBegin.y, _currentTrackBegin.z, _currentTrackPieceDirection });
 
+    const auto rideId = w->number;
     trackRemoveAction.SetCallback([=](const GameAction* ga, const GameActions::Result* result) {
```

Comment:
I think the previous issue also applies here, though it seems like this lambda has a lot more couplings.

Reply:
Quite a few arguments, so I think we'll leave that one. Normally with reviews this should sneak back in.

Reply:
I understand. Then I think this is good to go.